components/config-mgmt-service/integration_test/report_export_test.go

```diff
@@ -0,0 +1,226 @@
+	esClient := elastic.New(elasticsearchUrl)
+	c := config.New(esClient)
+	server := grpcserver.NewCfgMgmtServer(c)
+
+	lis := bufconn.Listen(1024 * 1024)
+	s := grpc.NewServer()
+	api.RegisterCfgMgmtServer(s, server)
+
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Fatalf("Server exited with error: %v", err)
+		}
+	}()
+
+	dialer := func(string, time.Duration) (net.Conn, error) { return lis.Dial() }
+
+	conn, err := grpc.DialContext(context.Background(), "bufnet", grpc.WithDialer(dialer), grpc.WithInsecure())
+	defer conn.Close()
+	require.NoError(t, err)
+
+	client := api.NewCfgMgmtClient(conn)
+
+	cases := []struct {
+		description     string
+		runsProjects    []string
+		allowedProjects []string
+		runIsReturned   bool
+	}{
+		{
+			description:     "Run is returned when the one run project matches the one allowed project",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{"one"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "No run is returned when the one run project does not match the one allowed project",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{"two"},
+			runIsReturned:   false,
+		},
+		{
+			description:     "Run is returned when the one run project matches one of the two allowed projects",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{"one", "two"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is returned when both of the two run projects match both of the allowed projects",
+			runsProjects:    []string{"one", "two"},
+			allowedProjects: []string{"one", "two"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is retuned when the run does not have an assigned project and the allowed project is unassigned",
+			runsProjects:    []string{},
+			allowedProjects: []string{authzConstants.UnassignedProjectID},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is returned when the run does not have an assigned project and no allowed project are provided",
+			runsProjects:    []string{},
+			allowedProjects: []string{},
+			runIsReturned:   true,
+		},
+		{
+			description:     "No run is returned when the run does not have an assigned project and one allowed project",
+			runsProjects:    []string{},
+			allowedProjects: []string{"one"},
+			runIsReturned:   false,
+		},
+		{
+			description:     "No run is returned when the run has one project and the allowed project is unassigned",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{authzConstants.UnassignedProjectID},
+			runIsReturned:   false,
+		},
+		{
+			description:     "Run is returned when the run has a matching project with one of the allowed project with unassigned",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{authzConstants.UnassignedProjectID, "one"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is returned when the run as a project and all projects are allowed",
```

Comment:
Something missing--maybe "as" should be "has" ?

Reply:
You are right, it was missing "has".